src/main/java/de/numcodex/feasibility_gui_backend/api/QueryHandlerRestController.java

```diff
@@ -31,9 +32,12 @@
 public class QueryHandlerRestController {
 
   private final QueryHandlerService queryHandlerService;
+  private final String apiBaseUrl;
 
-  public QueryHandlerRestController(QueryHandlerService queryHandlerService) {
+  public QueryHandlerRestController(QueryHandlerService queryHandlerService,
+      @Value("${app.apiBaseUrl}") String apiBaseUrl) {
     this.queryHandlerService = queryHandlerService;
+    this.apiBaseUrl = apiBaseUrl;
```

Comment:
`app.apiBaseUrl` does not have a default value and thus can be null. This should be avoided where possible.
So, we should either define an appropriate default value matching the implementation or at least introduce a non null check.

A check could look like this:
```
this.apiBaseUrl = Objects.requireNonNull(apiBaseUrl)
```

Reply:
Just saw the default value within the `application.yml`. A check would not hurt nonetheless 😃